Add Publishing your actor section

PerVillalva opened the pull request:
The purpose of this section is to offer a guide for developers to publish and monetize their code on the Apify platform.

@mstephen19 This is my first PR to the academy. I tried to maintain the same style and conventions I observed in the way you are structuring the content, but I might have missed some things. Let me know if there is anything wrong and I will promptly fix it. Thanks for the patience man. Cheers 🍻

mstephen19 commented on content/academy/apify_platform/publishing-your-actor.md:
```diff
@@ -0,0 +1,68 @@
+  - publishing-your-actor
+---
+
+# [](#publishing-actor) Publishing your actor
```
This file should really just be an introduction to the **Publishing your actor** section. Just a bit of an overview of what the section will cover. Maybe 50 lines absolute maximum imo

mstephen19 commented on content/academy/apify_platform/publishing-your-actor.md:
```diff
@@ -0,0 +1,68 @@
+
+## [](#readme) README
+
+The next step is to write a **README** detailing the actor’s features, reasons to scrape the targeted website, and an explanation about how to use the actor.
```
I don't think we should say "reasons to scrape the targeted website", because scrapers can be actors, but not ALL actors are scrapers. Actors can be anything!

I'm really trying to avoid labelling actors as JUST for scraping, so maybe it would be better to say something like "common use cases" or something like that.

mstephen19 commented on content/academy/apify_platform/publishing-your-actor.md:
```diff
@@ -0,0 +1,68 @@
+
+[https://github.com/zpelechova/readme-template](https://github.com/zpelechova/readme-template)
+
+Note that the complexity of the README should match the actor’s complexity. This means that the template above is not immutable and that you can adapt it to fit the particularities of your actor. You can check our guide, \***\*[How to write great README for your actors](https://help.apify.com/en/articles/2912548-how-to-write-great-readme-for-your-actors)\*\***, for more examples of README files.
```
We should cover a bit more on why it's important to write a readme. I think a bulleted list would work best.
Something like:
1. Users and other developers understand how to use your actor without needing to look at the code
2. People can understand why your actor even exists and what it does
3. Your actor can more easily be found on Google and other search engines
4.  etc. etc.

mstephen19 commented on content/academy/apify_platform/publishing-your-actor.md:
```diff
@@ -0,0 +1,68 @@
+
+## [](#make-your-actor-public) Make your actor Public
+
+Once you have finished coding and testing your actor, it's time to publish it.
```
Maybe link to a few resources on HOW to test an actor.

Link to this actor: https://apify.com/pocesar/actor-testing
And also to things like Jest testing library https://jestjs.io/

mstephen19 commented on content/academy/apify_platform/publishing-your-actor.md:
```diff
@@ -0,0 +1,68 @@
+
+![Actor settings](https://i.postimg.cc/qRz5jChr/actor-settings.png)
+
+Select the **Published** switch, and your actor should now be available to the public on Apify Store.
```
Make sure they configure the default memory and timeout for the actor before publishing. It defaults to 512MB which can be problematic and run into OOM errors (depending on the actor)

mstephen19 commented on content/academy/apify_platform/publishing_your_actor/monetizing-your-actor.md:
```diff
@@ -0,0 +1,88 @@
+
+![Apify billing](https://i.postimg.cc/NMqd7Pc6/Apify-Billing.jpg)
+
+## [](#price-and-trial-length) **Price and trial length**
```
This shouldn't be bold since it's already a heading

mstephen19 commented on content/academy/apify_platform/publishing_your_actor/monetizing-your-actor.md:
```diff
@@ -0,0 +1,88 @@
+
+**Don’t overlook promoting your actor**. This is arguably the most important step after publishing your tool. Taking advantage of all the possible mediums for promotion is what often differentiates very successful actors with dozens of paid users from actors with few to no users at all.
+
+Finally, don’t forget to check our [pro-tips](https://docs.apify.com/actors/publishing/seo-and-promotion#promotion) to take your actor promotion efforts to the next level.
```
Don't forget a nextup or I guess in this case, a wrap up

mstephen19 commented on content/academy/apify_platform/publishing_your_actor/monetizing-your-actor.md:
```diff
@@ -0,0 +1,88 @@
+menuWeight: 1
+category: courses
+paths:
+  - monetizing-your-actor
```
broken path should be apify_platform/publishing_your_actor/monetizing_your_actor

Btw, the MD files should be `snake_cased` instead of `using-dashes`

mstephen19 reviewed (changes requested):
Thanks so much @PerVillalva! The content itself looks great, there are just a few things I suggested to be added. Mainly, I think the content needs to be beefed up just a tad bit more and split into maybe 3-4 separate lessons.

The main issue is the build error that is happening. If you run `npm build` in the directory, you'll see the reasons it's failing and can fix them and run again to check if it still fails.

PerVillalva commented:
@mstephen19 Hey Matt, I struggled a little with finding all the broken paths, but it should be sorted out now 😅 

I added a brief overview section and fixed the deployment errors, but I won't have enough time to implement all the suggestions anytime soon. Can you take another look at the files to see if everything is correct? If there are no major mistakes, I believe we can go ahead and publish the sections and keep adding to the content in the future.

Cheers 🍻

mstephen19 commented:
Looks great to me! Thanks @PerVillalva. Let's get a second pair of eyes from @mnmkng

mnmkng commented on content/academy/apify_platform/publishing_actors_on_apify_store.md:
```diff
@@ -0,0 +1,21 @@
+
+# [](#publishing-actors-on-apify-store) Publishing actors on Apify Store
+
+[Apify Store](https://apify.com/store) is home to hundreds of public actors available to the Apify community. Anyone is welcome to publish actors in the store, and you can even [monetize your actors](https://get.apify.com/monetize-your-code) by renting it to users in the platform. 
```
```suggestion
[Apify Store](https://apify.com/store) is home to hundreds of public actors available to the Apify community. Anyone is welcome to publish actors in the store, and you can even [monetize your actors](https://get.apify.com/monetize-your-code) by renting them to users of the platform. 
```

mnmkng commented on content/academy/apify_platform/publishing_actors_on_apify_store/monetizing_your_actor.md:
```diff
@@ -0,0 +1,87 @@
+
+When publishing your actor on the Apify platform you have the choice to make it a **Paid actor** and get paid by the users benefiting from your tool.
+
+![Paid actor example](https://i.postimg.cc/k5VHybtm/scraper-example.png)
```
All images should be saved in the repo and minified using tinyPNG. See other lessons for examples how it's done.

mnmkng commented on content/academy/apify_platform/publishing_actors_on_apify_store/monetizing_your_actor.md:
```diff
@@ -0,0 +1,87 @@
+
+![Paid actor example](https://i.postimg.cc/k5VHybtm/scraper-example.png)
+
+Paid actor users pay a monthly rental fee to use the tool, which \***\*is **paid 100% to the actor’s developer.\*\* This means that, in the example above, the developer would get the full $45 monthly rent paid by each actor’s user.
```
```suggestion
Paid actor users pay a monthly rental fee to use the tool, which is **paid in full to the actor’s developer.** This means that, in the example above, the developer would get the full $45 monthly rent paid by each actor’s user.
```

mnmkng commented on content/academy/apify_platform/publishing_actors_on_apify_store/monetizing_your_actor.md:
```diff
@@ -0,0 +1,87 @@
+
+    While there are various ways to effectively promote your tool, it is usually best to share it on multiple channels. So, here are a few suggestions of channels you can use to promote your actor:
+
+    - **Reddit** **and Quora**
```
```suggestion
    - **Reddit and Quora**
```

mnmkng commented on content/academy/apify_platform/publishing_actors_on_apify_store/monetizing_your_actor.md:
```diff
@@ -0,0 +1,87 @@
+
+    Once your article is ready, you can check this list of relevant websites where you can publish your article: [hackernoon.com](https://hackernoon.com/), [techcrunch.com](https://techcrunch.com/), [news.ycombinator.com](https://news.ycombinator.com/), [dev.to](https://dev.to/), [producthunt.com,](https://www.producthunt.com/) [indiehackers.com](https://www.indiehackers.com/)
+
+**Don’t overlook promoting your actor**. This is arguably the most important step after publishing your tool. Taking advantage of all the possible mediums for promotion is what often differentiates very successful actors with dozens of paid users from actors with few to no users at all.
```
```suggestion
**Don’t overlook promoting your actor**. This is arguably the most important step after publishing your tool. Taking advantage of all the possible channels for promotion is what often differentiates very successful actors with dozens of paid users from actors with few to no users at all.
```

mnmkng reviewed (changes requested):
It's good content. Thanks Percy. Just a few nitpicks and suggestions.

mnmkng commented:
Ping @PerVillalva, there are still some unresolved comments.

PerVillalva commented:
Thanks for the reminder @mnmkng! 

I've resolved all comments. The only problem is that I cannot preview the images after I saved them in this repo and changed the paths, and I noticed that's also the case for the already published content. Is there anything I can do on my side to preview the images? Otherwise, everything should be ready. You can take another look to see if I wrote the paths correctly. Cheers 🍻
